docs/Readme.md

```diff
@@ -0,0 +1,80 @@
+# WiredCraft Back-end Developer Test
+
+## Table of Content
+* [Architecture](#architecture)
+  * [Technical Stack](#tech-stack)
+* [Design](#design)
+  * [Authentication](#authentication)
+    * [UML Diagram](#uml-sequence-diagram)
+    * [Authentication and Authorization Strategy](#authentication-and-authorization-strategy)
+  * [Logging](#logging)
+  * [Testing](#testing)
+* [Setup](#setup)
+* [Improvements](#improvements)
+
+## Architecture
+
+### Technical Stack
+
+- Framework -> Loopback 4
+  * I choose loopback 4 because of speed and it's extensibility. Loopback's codebase is well structured and saves you the headache of thinking about that. Also, for personal preference,
+  I lean towards strongly typed languages due to my experience with Go and Java.
+- Testing -> Mocha
+- Database -> MongoDB 4
+- Deployment -> Docker
+
+## Design
+
+### Authentication
+
+#### UML Sequence Diagram
+
+<p>Below is the UML sequence diagram for the user login flow. Assuming the user has already been created. </p>
+
+![uml](jwt-uml.png)
```

Comment:
what tools are used to draw this UML?
how can we update the UML if the interaction changes in the future?

Reply:
- I used [draw.io](https://app.diagrams.net/).
- Ideally with draw.io, I can save the file in Google Drive and edit it later.
   
Using a tool like [Mermaid](https://mermaid-js.github.io/mermaid/#/) or any text format UML tool would have been a better approach. This makes it easier to maintain. The scripted text can be included within the documentation and a tool/extension can be used for rendering.